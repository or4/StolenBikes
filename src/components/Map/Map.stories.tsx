import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, number } from '@storybook/addon-knobs';

import { Map } from './Map';

const stories = storiesOf('Map', module);
stories.addDecorator(withKnobs);

stories.add('Map', () => {
    return <Map coordinates={{ lng: getKnobsLng(), lat: getKnobsLat() }} />;
});

function getKnobsLng() {
    return number('lng', 13.4275004);
}

function getKnobsLat() {
    return number('lat', 52.4890809);
}
