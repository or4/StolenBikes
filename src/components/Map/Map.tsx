import React from 'react';
import styled from 'styled-components';
import GoogleMap from 'google-map-react';

import { IGeoCoordinates } from 'types';
import { media } from 'ui/media';

interface IProps {
    coordinates: IGeoCoordinates;
}

const Container = styled.div`
    ${media.mobile`
		> div {
			padding-top: 100% !important;
		}
  	`}

    ${media.tabletNDesktop`
		> div {
			padding-top: 50% !important;
		}
	`}
`;

const Marker = styled.div`
    border-radius: 50%;
    background-color: rgba(0, 0, 0, 0.4);
    width: 20px;
    height: 20px;
`;

export const Map: React.FC<IProps> = ({ coordinates }: IProps) => {
    // @ts-ignore
    const marker = <Marker {...coordinates} text="" />;

    return (
        <Container>
            <GoogleMap
                defaultZoom={15}
                bootstrapURLKeys={{
                    key: 'AIzaSyDaxTbb824BrjJMUfBeCc5Cxv6sbuwYnhs',
                }}
                center={coordinates}
            >
                {marker}
            </GoogleMap>
        </Container>
    );
};
