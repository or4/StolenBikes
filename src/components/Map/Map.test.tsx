import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import { Map } from './Map';

describe('Map', () => {
    test('should render correctly', () => {
        const container = shallow(<Map coordinates={{ lng: 13.4275004, lat: 52.4890809 }} />);
        expect(container).toMatchSnapshot();
    });
});
