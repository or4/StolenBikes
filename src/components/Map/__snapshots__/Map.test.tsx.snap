// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map should render correctly 1`] = `
<styled.div>
  <GoogleMap
    bootstrapURLKeys={
      Object {
        "key": "AIzaSyDaxTbb824BrjJMUfBeCc5Cxv6sbuwYnhs",
      }
    }
    center={
      Object {
        "lat": 52.4890809,
        "lng": 13.4275004,
      }
    }
    debounced={true}
    defaultZoom={15}
    distanceToMouse={[Function]}
    googleMapLoader={[Function]}
    heatmap={Object {}}
    heatmapLibrary={false}
    hoverDistance={30}
    layerTypes={Array []}
    options={[Function]}
    style={
      Object {
        "height": "100%",
        "margin": 0,
        "padding": 0,
        "position": "relative",
        "width": "100%",
      }
    }
    yesIWantToUseGoogleMapApiInternals={false}
  >
    <styled.div
      lat={52.4890809}
      lng={13.4275004}
      text=""
    />
  </GoogleMap>
</styled.div>
`;
